ui/src/utils/groupByTimeSeriesTransform.ts

```diff
@@ -246,13 +246,34 @@ const constructCells = (
           })
         }
       } else {
-        metaQuerySeries = [columns, ...values]
         isMetaQuery = true
-        labels = columns.map(c => ({
-          label: c,
-          responseIndex,
-          seriesIndex,
-        }))
+
+        if (serieses.length === 1) {
+          labels = columns.map(c => ({
+            label: c,
+            responseIndex,
+            seriesIndex,
+          }))
+
+          metaQuerySeries = [columns, ...values]
+        } else {
+          labels = ['measurement', columns[0]].map(c => ({
```

Comment:
It looks like this will need to be adapted to handle multiple columns. Running the query `SHOW FIELD keys on telegraf` returns `fieldKey` and `fieldType` with multiple series.

Reply:
It looks like we are also recreating the same labels on each iteration over a series and we might be able to just do this on `ind === 0`.

Reply:
yeah, this whole thing is within a `forEach` so there's some redundancy